src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,73 @@
+package duke;
+
+public class Parser {
+
+    /**
+     * Parses input by the user and returns the corresponding Command object.
+     *
+     * @param command String input by the user.
+     * @return Command object to be executed.
+     * @throws DukeException If command cannot be parsed into a known command.
+     */
+    public static Command parse(String command) throws DukeException {
+        if (command.equals("list")) {
```

Comment:
You can use `String.split(" ")` at the start, and use `command[0]` to handle different types of command. 

Subsequently, you can use the respective methods to handle the validity of the command.